test/integration/Select.spec.js

```diff
@@ -0,0 +1,41 @@
+// @flow weak
+
+import React from 'react';
+import { assert } from 'chai';
+import { ReactWrapper } from 'enzyme';
+import { createMount } from 'src/test-utils';
+import SelectAndDialog from './fixtures/select/SelectAndDialog';
+
+describe('<Select> integration', () => {
+  let mount;
+
+  before(() => {
+    mount = createMount();
+  });
+
+  after(() => {
+    mount.cleanUp();
+  });
+
+  describe('with Dialog', () => {
+    let wrapper;
+
+    before(() => {
+      wrapper = mount(<SelectAndDialog />);
+    });
+
+    it('should focus the first item as nothing has been selected', () => {
+      const portal = wrapper.find('Modal').node.mountNode.firstChild;
+      const portalWrapper = new ReactWrapper(portal, portal);
+
+      const selectDisplay = portalWrapper.find('[data-mui-test="SelectDisplay"]');
+      selectDisplay.simulate('click');
+      const menuEl = document.querySelector('[data-mui-test="Menu"]');
+      assert.strictEqual(
+        document.activeElement,
+        menuEl && menuEl.children[1],
+        'should be the 2nd menu item',
+      );
+    });
+  });
+});
```

Comment:
What do you think about adding test from #8169 ? Btw looks like it isn't fixed by this PR, I tested it locally and problem remains.

Reply:
The focus is supposed to go back to the Select once an option is selected. This fix is about the wrong teardown focus behavior as well as the broken keyboard navigation.

Reply:
It's all good on my end.

Reply:
But this test case is also supposed to check that focus goes back to the `Select`, isn't it?

Reply:
We already have test for this behavior. As far as I know, stacking modals doesn't change the behavior hence left untested here.

Reply:
Added a [comment](https://github.com/callemall/material-ui/issues/8169#issuecomment-329964422) to related issue